### DISCUSSION TITLE
Discuss Data Provenance and Administrative Information in Library Linked Data: Reviewing RDA in RDF, BIBFRAME, and Wikidata

### DISCUSSION THREAD

**Discussion by briesenberg07** [crystalyragui/MARC2RDA#391] [Announcements]
> Full title: Data Provenance and Administrative Information in Library Linked Data: Reviewing RDA in RDF, BIBFRAME, and Wikidata
> Author: Shoichi Taniguchi
> https://doi.org/10.1080/01639374.2023.2178048

Please indicate your availability in the form at URL:  
**https://forms.office.com/r/QHL04wtbKh**

@gerontakos 
@CECSpecialistI 
@GordonDunsire 
@junghaelee 
@lake44me 
@JianPLee 
@SitaKB

**Comment by GordonDunsire** [comment]
Colleagues  
I don't have access to the journal, so can someone email me a download for review?  
On 05/05/2023 01:00 BST Benjamin Riesenberg ***@***.***> wrote:
> Full title: Data Provenance and Administrative Information in Library Linked Data: Reviewing RDA in RDF, BIBFRAME, and Wikidata
> Author: Shoichi Taniguchi
> https://doi.org/10.1080/01639374.2023.2178048 

   Cheers 
   Gordon

**Comment by SitaKB** [comment]
[Data Provenance and Administrative Information in Library Linked Data Reviewing RDA in RDF BIBFRAME and Wikidata.pdf](https://github.com/uwlib-cams/MARC2RDA/files/11406335/Data.Provenance.and.Administrative.Information.in.Library.Linked.Data.Reviewing.RDA.in.RDF.BIBFRAME.and.Wikidata.pdf)

**Comment by briesenberg07** [comment]
Taniguchi creates a bibliographic description set within a named graph, then uses the graph IRI as IRI for metadata Work:
```
# TriG RDF Dataset Language
@prefix dataset: <http://example.org/dataset/> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdac: <http://rdaregistry.info/Elements/c/> .
@prefix rdaw: <http://rdaregistry.info/Elements/w/> .

# ADAPTED from Taniguchi FIG 15
    # unsure how this model could include a metadata Manifestation

GRAPH dataset:graph001tan
{
    dataset:resource001tan
        rdf:type rdac:C10001 ;
        rdaw:P10223 "Go down, Moses"@en ; # preferred title of work
        rdaw:P10219 "1942"^^<http://id.loc.gov/datatypes/edtf> ; # date of work
        # ...additional statements on Work entity
        rdaw:P10629 dataset:graph001 . # work described with metadata by 
}

dataset:graph001tan
    rdf:type rdac:C10001 ;
    # ...additional statements on metadata work
    # Does Taniguchi's FIG 15 allow for existence of Manifestation of metadata work?
    rdaw:P10621 dataset:resource001tan . # is metadata description of work
```
LIKELY proposal for Sinopia RDA/RDF data provenance templates would mint a metadata Work IRI, this would not be a graph IRI
```
dataset:resource001uw
    rdf:type rdac:C10001 ;
        rdaw:P10223 "Go down, Moses"@en ; # preferred title of work 
        rdaw:P10219 "1942"^^<http://id.loc.gov/datatypes/edtf> ; # date of work
        # ...additional statements on Work entity
        rdaw:P10629 dataset:resource002uw . # work described with metadata by 

dataset:resource002uw
    rdf:type rdac:C10001 ;
    # ...additional statements on metadata work
    # has (metadata) manifestation of work ...
    rdaw:P10621 dataset:resource001uw . # is metadata description of work
```
Taniguchi says:
> [...] a little difficulty in the semantic representation of meta-level information – the nodes of provenance and administrative information are connected to the target resource itself, not metadata for the target resource.

The "nodes of provenance and administrative information [being] connected to the target resource" seems to be a problem for Taniguchi. Does my proposal above have this problem?

**Reply by pan-zhuo** [reply]
My concern about using named graphs:
RDF graphs are supposed to be static. A change to a graph will result in a new graph.

What happens when a metadata work changes?
If it's a new work then the graph IRI should change as well.
If it's not a new work (diachronic/integrating work) then different serializations should realize different works due to the WEM lock...

**Reply by pan-zhuo** [reply]
> The "nodes of provenance and administrative information [being] connected to the target resource" seems to be a problem for Taniguchi. Does my proposal above have this problem?

I think this proposal does not have "nodes of provenance and administrative information" in the sense defined by Taniguchi. It has a "Work" node. It seems that "Work" nodes and "nodes of provenance and administrative information" are different things according to Taniguchi, as shown in Pattern 7.

**Reply by GordonDunsire** [reply]
Using an instance of Work as the target resource gets confused with an instance of metadata Work that describes the target resource. So I've recast @briesenberg07's examples as a manifestation description set and its metadata work.

TAN:

GRAPH dataset:metadataWork001tan
{
 dataset:manifestation001tan
  rdam:P30156 "My manifestation" ; # title proper, for RDA conformance
  #additional statements on Manifestation instance
  rdam:P30139 ex:expression001tan . # expression manifested, for RDA conformance
}

dataset:metadataWork001tan
  rdaw:P10002 "dataset:metadataWork001tan" ; # identifier for work, for RDA conformance
  rdaw:P10072 dataset:metadataManifestation001tan ; # manifestation of metadata work, for RDA conformance - this may be something that is not desirable or necessary
  #additional statements on metadata Work instance
  rdaw:P10617 dataset:manifestation001tan  . # metadata description of manifestation
  
Question gd1: should the inverse statement that relates the manifestation being described with the metadata work be included in the GRAPH declaration, or excluded because it is self-referential***?  
  
    rdam:P30462 dataset:metadataWork001tan . # manifestation described with metadata by 
	
UW:

dataset:manifestation001uw
  rdam:P30156 "My manifestation" ; # title proper, for RDA conformance
  #additional statements on Manifestation instance
  rdam:P30139 ex:expression001uw ; # expression manifested, for RDA conformance
  rdam:P30462 dataset:metadataWork001tan . # manifestation described with metadata by

dataset:metadataWork001uw
  rdaw:P10002 "dataset:metadataWork001uw" ; # identifier for work, for RDA conformance
  rdaw:P10072 dataset:metadataManifestation001uw ; # manifestation of metadata work, for RDA conformance - this may be something that is not desirable or necessary
  #additional statements on metadata Work instance
  rdaw:P10617 dataset:manifestation001uw  . # metadata description of manifestation
  
The UW proposal resolves question gd1.

Question gd2: should we ask RSC to relax the conformance requirements for metadata works, so that we don't have to generate a metadata expression or manifestation if there's nothing to say about them. The current RDA instructions are intended to be refined when communities of practice mature.

I share @pan-zhuo's concerns about named graphs. The RDA approach is compatible with data provenance for single metadata statements; this is automatically implemented and managed in a typical quad store. The RDA approach is also compatible with data provenance for fixed/static metadata description sets which might be implemented as "core" descriptions in an application. But further development of the concept of "integrating aggregating metadata work" in RDA is required before an operational method of using named dynamic graphs can be developed.

If we decide that the transform is one-off, then the transform output might be considered as static description sets.

I think provenance and administrative data are clearly separated by the RDA properties:

dataset:metadataWork001uw
  #administrative metadata
  rdaw:P10065 "UW project" ; # creator agent of work
  rdaw:P10219 "2023" ; # date of work
  #provenance metadata
  rdaw:P10406 dataset:manifestation001uw ; # source consulted; the manifestation itself
  rdaw:P10406 "{MARC 21 record in exchange format}" . # source consulted; the MARC 21 record
  
*** I think we may encounter a few quasi [strange loops](https://en.wikipedia.org/wiki/Strange_loop) due to self-referentiality ...

[ps I may be late in joining the zoom discussion today, due to the RSC's "public session"]

**Comment by briesenberg07** [comment]
# Discussion (partial notes)

- RIMMF 6 will have some functionality for data provenance, details coming soon
- Description-set level metadata work is suitable for *static* metadata works
- Taniguchi makes an aside that he doesn't understand collection-level description
- RDA places metadata works in the context of other works which describe other works
- Article pointed out lack of examples; let's share our work with the RSC 
   - It's an undeveloped model, RSC wants feedback and for the community to make some decisions about implementation
- Minimal-description requirements might be reviewed for metadata Works
- What kinds of data provenance is required?
	- source recorded: this really can only exist at the statement level
	- other things like cataloging rules, author of metadata, could be recorded at description-set level 
- Would it be reasonable to ask Sinopia developers to add a feature: turn on or off the ability to create a new IRI in a nested-resource property template? This would make RDA/RDF in Sinopia so much easier
- Current Sinopia template work is focusing on description-set level data provenance, putting statement-level provenance off to the side for the time being (no mechanism to reify triples in Sinopia--and do we want to add this kind of complexity??)
- What about creating additional data-provenance examples--like statement-level data-provenance--in data external to Sinopia, and sharing these with the RSC?
	- @gerontakos and @CECSpecialistI are interested in this
- Question about Nomens for work groups
	- If all works in the Work group share an appellation, wouldn't it make sense to mint one Nomen for them?
	- Entity boundary requires that Nomens with the same Nomen string which refer to different works must be different Nomen entities; need follow-up on this topic

**Reply by GordonDunsire** [reply]
To clarify the limitations of the "uw" proposal (my version):

dataset:metadataWork001uw does not explicitly state what data is being described for provenance. It is assumed that it is describing the description set dataset:manifestation001uw that it links with rdaw:P10617 "metadata description of manifestation". This is why it can only be applied to static description sets.

Furthermore, the static description sets should consist of the same pattern of statements; for example, there is always a title, date, and publisher, with values that vary from instance to instance.

An associated limitation is that only one metadata work can be 'active' for the provenance of any entity instance. Otherwise, the metadata work itself needs provenance, for example when provenance from different agencies is merged if the static description set is de-duplicated in a consortium application.

This is typical of "record"-based cataloguing in a closed world application.

I see only two methods of overcoming this limitation, say in an open world application using the Open World Assumption:

1. The "tan" version which reifies the graph of dataset:manifestation001tan as dataset:metadataWork001tan.
2. The Wikidata approach, which duplicates the relevant statements of dataset:manifestation001tan within dataset:metadataWork001tan.

1. is the recommendation of the RDA Toolkit guidance, but 2. is not excluded.

I now cannot see where I thought Tanaguchi was misunderstanding the recent development of collection description in RDA, but I think it would have been worth his mentioning or discussing the impact of dynamic data on provenance. This is not the same as the discussion on time-limited data such as Adams' attendance at university. Instead, it is what I was alluding to with "integrating aggregating metadata work".

The RDA definition of "metadata description set" is "A metadata work that is an aggregating work that is a plan to select and arrange one or more expressions of one or more metadata statements that describe one or more individual instances of RDA entities and embody them in an aggregate."

That is, dataset:metadataWork001 is an aggregating work, which is a plan to gather a set of metadata statements - and is NOT the set of metadata statements itself. The set of statements, along with the aggregating work that selects and organizes them, is embodied in an aggregate manifestation. An aggregate metadata manifestation may be an RDF graph in a specific serialization such as dataset:manifestation001 in turtle with carrier "online resource", a MARC 21 "record", a catalogue (describing a collection Work) or a bibliography, filmography, or discography (describing an aggregating work).

This is recursive, in the sense that the description set for dataset:metadataWork001 is itself a metadata work:

GRAPH dataset:metametadataWork001tan
{
dataset:metadataWork001tan
rdaw:P10002 "dataset:metadataWork001tan" ;
rdaw:P10072 dataset:metadataManifestation001tan ;
#additional statements on metadata Work instance
rdaw:P10617 dataset:manifestation001tan .
}

What happens when the catalogue, graph, or record is amended? A new aggregate is created, requiring a new metadata Work - but the catalogue, etc. is a manifestation of a diachronic work, specifically an integrating work, "A diachronic work that is planned to be realized by a single expression". The single expression can only be the aggregating expression ... This requires clarification in RDA, and I have been discussing a possible resolution with Deborah Fritz (as a result of the RIMMF work).

Taniguchi does not mention the subject relationship hierarchy in RDA, perhaps to avoid confusion with "subject" in the context of RDF triples. The hierarchy is:

Manifestation subject manifestation of Work
> manifestation described by
>> manifestation described with metadata by
>>> finding aid
>>>> catalogue

I agree with Taniguchi that "It is not clear what side effects this recursive-like complex configuration may have", but the scope note for "metadata description set" is "The plan may be part of an application profile" and I think this where the effects of recursion should (eventually) be managed. 

My head is now recursing, so I'll stop :-)

**Reply by GordonDunsire** [reply]
Please note that I opened a new [discussion on nomens and work groups](https://github.com/uwlib-cams/MARC2RDA/discussions/428).

**Comment by lake44me** [comment]
First I apologize for not responding to the Doodle poll, I was so busy during these last 2 weeks that I am now catching up with emails, and I couldn't have made any of the dates anyway.  Perhaps it is better, I will read the article and the comments and then IF I have anything to add, add it.  

Meanwhile, someone called my attention to this W3C blog post 
  W3C is starting to draft updates to RDF: [https://www.w3.org/blog/news/archives/9906]. 

I have just started poking around in these, and noticed the introduction of "quoted triples". 
[https://www.w3.org/TR/2023/WD-rdf12-concepts-20230516/#dfn-quoted-triple]

 I'm not seeing an example of quoted triple syntax in this document.  This concept was introduced in the RDF-Star and SPARQL-star community report in 2021  https://www.w3.org/2021/12/rdf-star.html#quoted-triples so you may be familiar with it.  There are examples there (using double angle brackets to enclose the quoted triple).  
 
 Thought this might be interesting.  This is first-draft new standard version, so may take awhile to become part of the standard.  Wondering if it could help us with "reification" issues, referring to data sources at the statement level, etc.  Or not.
 
 Each document seems to have a section identifying what's new in the 1.2 version.
